### ISSUE TITLE
TF golang client memory leak

### ISSUE THREAD

**Issue by iammeizu** [tensorflow/tensorflow#57356]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 1 and tf 2

### Custom Code

Yes

### OS Platform and Distribution

ubuntu20.04

### Python version

3.8

### GCC/Compiler version

9.3.0

### CUDA/cuDNN version

11.4

### Current Behaviour?

```shell
Action1: new a tensor and use the only one tensor to do session run will not result in memory raise.
Action2: new a tensor every time before seesion run will lead to memory rasie.

Conclusion: NewTensor api lead to memory leak

Suggestion: Release tensor obvisly instead of just set finializer
```

### Standalone code to reproduce the issue

```shell
package main

import (
	"flag"
	"fmt"
	"io/ioutil"
	"log"

	tf "github.com/tensorflow/tensorflow/tensorflow/go"
)

func main() {
	// An example for using the TensorFlow Go API for image recognition
	// using a pre-trained inception model (http://arxiv.org/abs/1512.00567).
	//
	// Sample usage: <program> -dir=/tmp/modeldir -image=/path/to/some/jpeg
	//
	// The pre-trained model takes input in the form of a 4-dimensional
	// tensor with shape [ BATCH_SIZE, IMAGE_HEIGHT, IMAGE_WIDTH, 3 ],
	// where:
	// - BATCH_SIZE allows for inference of multiple images in one pass through the graph
	// - IMAGE_HEIGHT is the height of the images on which the model was trained
	// - IMAGE_WIDTH is the width of the images on which the model was trained
	// - 3 is the (R, G, B) values of the pixel colors represented as a float.
	//
	// And produces as output a vector with shape [ NUM_LABELS ].
	// output[i] is the probability that the input image was recognized as
	// having the i-th label.
	//
	// A separate file contains a list of string labels corresponding to the
	// integer indices of the output.
	//
	// This example:
	// - Loads the serialized representation of the pre-trained model into a Graph
	// - Creates a Session to execute operations on the Graph
	// - Converts an image file to a Tensor to provide as input to a Session run
	// - Executes the Session and prints out the label with the highest probability
	//
	// To convert an image file to a Tensor suitable for input to the Inception model,
	// this example:
	// - Constructs another TensorFlow graph to normalize the image into a
	//   form suitable for the model (for example, resizing the image)
	// - Creates and executes a Session to obtain a Tensor in this normalized form.
	modeldir := flag.String("dir", "", "Directory containing the trained model files. The directory will be created and the model downloaded into it if necessary")
	//imagefile := flag.String("image", "", "Path of a JPEG-image to extract labels for")
	flag.Parse()
	if *modeldir == "" {
		flag.Usage()
		return
	}
	// Load the serialized GraphDef from a file.
	// modelfile, labelsfile, err := modelFiles(*modeldir)
	//if err != nil {
	//	log.Fatal(err)
	//}
	model, err := ioutil.ReadFile(*modeldir)
	if err != nil {
		log.Fatal(err)
	}

	// Construct an in-memory graph from the serialized form.
	graph := tf.NewGraph()
	if err := graph.Import(model, ""); err != nil {
		log.Fatal(err)
	}

	// Create a session for inference over graph.
	session, err := tf.NewSession(graph, nil)
	if err != nil {
		log.Fatal(err)
	}
	defer session.Close()

	// Run inference on *imageFile.
	// For multiple images, session.Run() can be called in a loop (and
	// concurrently). Alternatively, images can be batched since the model
	// accepts batches of image data as input.
	//tensor, err := makeTensorFromImage(*imagefile)
	//if err != nil {
	//	log.Fatal(err)
	//}

	inputs := make([]string, 0)
	for i := 0; i< 8;i++ {
		inputs = append(inputs, "测试文本输入")
	}

	tensor, err := tf.NewTensor(inputs)
	if err != nil {
		log.Fatal(err)
	}
	for i := 0; i < 1000000; i++ {
		output, err := session.Run(
			map[tf.Output]*tf.Tensor{
				graph.Operation("input_tensor_name").Output(0): tensor,
			},
			[]tf.Output{
				graph.Operation("output_tensor_name").Output(0),
			},
			nil)
		if err != nil {
			log.Fatal(err)
		}
		probabilities := output[0].Value().([][]float32)[0]
		if i % 1000 == 0 {
			fmt.Println("repeat ", probabilities)
		}
	}
   	// output[0].Value() is a vector containing probabilities of
	// labels for each image in the "batch". The batch size was 1.
	// Find the most probably label index.
	//printBestLabel(probabilities, labelsfile)
}
```

**Comment by iammeizu** [comment]
![1661066879182](https://user-images.githubusercontent.com/18156457/185780528-6f920a6d-128f-415c-8279-d50e3cf85a0c.png)
Action1：right red box

**Comment by sushreebarsa** [comment]
@iammeizu Could you please confirm if this issue is replicating in TF v2.9.0 ?
Thank you!

**Comment by wamuir** [comment]
If TF < 2.5, there was indeed a memory leak associated with deallocs of string tensors in, fixed by #49925.
